modP2pImpl/src/org/aion/p2p/impl1/tasks/TaskWrite.java

```diff
@@ -92,25 +92,50 @@ public void run() {
             }
             buf.flip();
 
+            long t1 = System.currentTimeMillis(), t2;
+            int wrote = 0;
             try {
-                while (buf.hasRemaining()) {
-                    // @Attention:  very important sleep , otherwise when NIO write buffer full,
-                    // without sleep will hangup this thread.
-                    Thread.sleep(0, 1);
-                    sc.write(buf);
+                do {
+                    int result = sc.write(buf);
+                    wrote += result;
+
+                    if (result == 0) {
+                        // @Attention:  very important sleep , otherwise when NIO write buffer full,
+                        // without sleep will hangup this thread.
+                        Thread.sleep(0, 1);
+                    }
+
+                    t2 = System.currentTimeMillis() - t1;
+                } while (buf.hasRemaining() && (t2 < 100));
+
+                if (p2pLOG.isTraceEnabled() && (t2 > 10)) {
+                    p2pLOG.trace(
+                            "msg write: id {} size {} time {}ms length {}",
```

Comment:
minor update: `time {} ms` instead of `time {}ms`

Reply:
updated